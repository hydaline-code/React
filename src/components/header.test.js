import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import Header from './header';

it('should navigate to the home page when the Home link is clicked', () => {
  const { getByText } = render(
    <Router>
      <Header />
    </Router>,
  );

  const homeLink = getByText('Home |');
  fireEvent.click(homeLink);

  expect(window.location.pathname).toBe('/');
});

it('should navigate to the calculator page when the Calculator link is clicked', () => {
  const { getByText } = render(
    <Router>
      <Header />
    </Router>,
  );

  const calculatorLink = getByText('Calculator |');
  fireEvent.click(calculatorLink);

  expect(window.location.pathname).toBe('/calculator');
});

it('should navigate to the quotes page when the Quotes link is clicked', () => {
  const { getByText } = render(
    <Router>
      <Header />
    </Router>,
  );

  const quotesLink = getByText('Quotes |');
  fireEvent.click(quotesLink);

  expect(window.location.pathname).toBe('/quotes');
});
