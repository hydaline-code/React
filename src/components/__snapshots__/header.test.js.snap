// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header snapshot test 1`] = `
<header
  className="header"
>
  <div
    className="logo"
  >
    <img
      alt="Logo"
      className="logo"
      src="logo.png"
    />
    <h4>
      Maths  Magicians
    </h4>
  </div>
  <nav>
    <ul
      className="nav-menu"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home |
        </a>
      </li>
      <li>
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator |
        </a>
      </li>
      <li>
        <a
          href="/quotes"
          onClick={[Function]}
        >
          Quotes |
        </a>
      </li>
    </ul>
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-bars menu"
      data-icon="bars"
      data-prefix="fas"
      focusable="false"
      onClick={[Function]}
      role="img"
      style={Object {}}
      viewBox="0 0 448 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 96C0 78.3 14.3 64 32 64H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32C14.3 128 0 113.7 0 96zM0 256c0-17.7 14.3-32 32-32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32-14.3-32-32zM448 416c0 17.7-14.3 32-32 32H32c-17.7 0-32-14.3-32-32s14.3-32 32-32H416c17.7 0 32 14.3 32 32z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
  </nav>
</header>
`;
